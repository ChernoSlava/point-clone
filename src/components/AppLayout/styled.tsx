import styled from 'styled-components';

export const AppStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  gap: 10px;
  height: 100%;
  padding: 20px;
  margin: 0 auto;
  background: #F5F5F5;
  background-color: rgb(255, 255, 255);
`;

export const MainLogoStyled = styled.img`
  margin: 45px 0;
  width: 170px;
  height: 48px 
`;
