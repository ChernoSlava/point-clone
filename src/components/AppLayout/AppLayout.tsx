import React from "react";
import { Link, Outlet } from "react-router-dom";
import { LOGO_URL } from "@constants/index";

import { AppStyled, MainLogoStyled } from "./styled";

const AppLayout: React.FC<{
  children: React.ReactElement | null;
}> = ({ children }): React.JSX.Element => {
  const logoLink = process.env.REACT_APP_LOGO_LINK || LOGO_URL;

  return (
    <AppStyled>
      <Link to="/">
        <MainLogoStyled src={logoLink} alt="Point logo" />
      </Link>
      <Outlet />
      {children}
    </AppStyled>
  );
};

export { AppLayout };
