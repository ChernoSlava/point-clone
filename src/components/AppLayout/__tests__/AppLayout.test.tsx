/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { MemoryRouter } from "react-router-dom";
// eslint-disable-next-line import/no-extraneous-dependencies
import { cleanup, render } from "@testing-library/react";

import { AppLayout } from "../AppLayout";

describe("AppLayout tests", () => {
  afterEach(() => {
    cleanup();
  });

  test("renders without children", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <AppLayout logo="test" />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  test("renders with children", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <AppLayout logo="test">
          <div>foo</div>
        </AppLayout>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    unmount();
  });
});
