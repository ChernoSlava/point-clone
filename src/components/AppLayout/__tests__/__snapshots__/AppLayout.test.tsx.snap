// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout tests renders with children 1`] = `
<div>
  <div
    class="sc-beySPh bqUtuk"
  >
    <a
      href="/"
    >
      <img
        alt="Point logo"
        class="sc-guDLey SRfIm"
        src="test"
      />
    </a>
    <div>
      foo
    </div>
  </div>
</div>
`;

exports[`AppLayout tests renders without children 1`] = `
<div>
  <div
    class="sc-beySPh bqUtuk"
  >
    <a
      href="/"
    >
      <img
        alt="Point logo"
        class="sc-guDLey SRfIm"
        src="test"
      />
    </a>
  </div>
</div>
`;
